InteractionBase/src/vis/ViewSwitcherMenu.cpp

```diff
@@ -0,0 +1,141 @@
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+#include "ViewSwitcherMenu.h"
+
+#include "nodes/ViewSwitcherEntry.h"
+#include "vis/VViewSwitcherEntry.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "VisualizationBase/src/cursor/TextCursor.h"
+
+namespace Interaction {
+
+ITEM_COMMON_DEFINITIONS(ViewSwitcherMenu, "item")
+
+ViewSwitcherMenu* ViewSwitcherMenu::instance{};
+
+void ViewSwitcherMenu::show(QVector<Model::Node*> selectableNodes, Visualization::Item* target)
+{
+	QApplication::postEvent(target->scene(),
+							new Visualization::CustomSceneEvent( [=]() { showNow(selectableNodes, target); }));
+}
+
+void ViewSwitcherMenu::hide()
+{
+	if (instance)
+		QApplication::postEvent(instance->scene(),
+								new Visualization::CustomSceneEvent( [&]() { hideNow(); }));
+}
+
+void ViewSwitcherMenu::showNow(QVector<Model::Node*> selectableNodes, Visualization::Item* target)
+{
+	ViewSwitcherMenu::hideNow();
+	instance = new ViewSwitcherMenu(selectableNodes, target);
+	target->scene()->addTopLevelItem(instance);
+	instance->installSceneEventFilter(instance);
+	instance->setFiltersChildEvents(true);
+}
+
+void ViewSwitcherMenu::hideNow()
+{
+	if (instance)
+	{
+		instance->removeSceneEventFilter(instance);
+		instance->scene()->removeTopLevelItem(instance);
+	}
+	SAFE_DELETE_ITEM(instance);
+	instance = nullptr;
+}
+
+ViewSwitcherMenu::ViewSwitcherMenu(QVector<Model::Node*> selectableNodes,
+											 Visualization::Item* target, StyleType* style)
+	: Super(selectableNodes, target, style)
+{
+}
+
+bool ViewSwitcherMenu::sceneEventFilter(QGraphicsItem *watched, QEvent *event)
+{
+	if (inEditMode_ && event->type() == QEvent::KeyPress)
+	{
+		//If editing, let the mover use the cursor keys to move within the item itself,
+		//rather than between items
+		auto keyEvent = static_cast<QKeyEvent*>(event);
+		if (keyEvent->key() == Qt::Key_Left || keyEvent->key() == Qt::Key_Right)
+			return false;
+	}
+	auto result = Super::sceneEventFilter(watched, event);
+	if (result)
+		return true;
+	else if (event->type() == QEvent::KeyPress)
+	{
+		//F2 can trigger a rename of the view item,
+		//also makes sure left click and enter then don't actually
+		//execute the existing item to make editing easier
+		auto keyEvent = static_cast<QKeyEvent*>(event);
+		auto asSwitcher = DCast<VViewSwitcherEntry>(focusedItem());
+		if (asSwitcher && (keyEvent->key() == Qt::Key_F2
+				|| (keyEvent->key() == Qt::Key_Return && inEditMode_)))
+		{
+			inEditMode_ = !inEditMode_;
+			if (!inEditMode_)	selectItem(asSwitcher);
+			else
+			{
+				asSwitcher->setGraphicsEffect(nullptr);
```

Comment:
I feel like the next 3 lines are the only ones that somehow truly belong to the ViewSwitcher. All the rest feels generic and should somehow make it in the base Menu. I am also willing to help with this, since I do see the need to merge all of these with AutoComplete.